experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -0,0 +1,493 @@
+// set(key: FieldKey, content: FlexibleFieldContent<TSchema["mapFields"]>): void;
+// {
+// 	readonly [key in keyof TFields as `set${Capitalize<key & string>}`]: (
+// 		content: FlexibleFieldContent<TFields[key]>,
+// 	) => void;
+// };
+// This could be enabled to allow assignment via `=` in some cases.
+// & {
+// 	// Setter properties (when the type system permits)
+// 	[key in keyof TFields]: UnwrappedField<TFields[key]> & StructSetContent<TFields[key]>;
+// }
+
+// #endregion
+
+// #region Field Kinds
+
+/**
+ * Strongly typed tree literals for inserting as the content of a field.
+ * @alpha
+ */
+export type FlexibleFieldContent<TSchema extends FieldSchema> = SchemaAware.TypedField<
+	TSchema,
+	SchemaAware.ApiMode.Flexible
+>;
+
+/**
+ * Strongly typed tree literals for inserting as a node.
+ * @alpha
+ */
+export type FlexibleNodeContent<TTypes extends AllowedTypes> = SchemaAware.AllowedTypesToTypedTrees<
+	SchemaAware.ApiMode.Flexible,
+	TTypes
+>;
+
+/**
+ * Field this stores a sequence of children.
+ *
+ * @remarks
+ * Allows for concurrent editing based on index, adjusting the locations of indexes as needed so they apply to the same logical place in the sequence when rebased and merged.
+ * @alpha
+ */
+export interface Sequence<TTypes extends AllowedTypes> extends UntypedField {
+	/**
+	 * Gets a node of this field by its index with unboxing.
+	 * Note that a node must exist at the given index.
+	 */
+	at(index: number): UnboxNodeUnion<TTypes>;
+
+	/**
+	 * Gets a boxed node of this field by its index.
+	 * Note that a node must exist at the given index.
+	 */
+	boxedAt(index: number): TypedNodeUnion<TTypes>;
+
+	/**
+	 * Calls the provided callback function on each child of this sequence, and returns an array that contains the results.
+	 * @param callbackfn - A function that accepts the child, its index, and this field.
+	 */
+	map<U>(callbackfn: (value: UnboxNodeUnion<TTypes>, index: number, array: this) => U): U[];
+
+	/**
+	 * Calls the provided callback function on each child of this sequence, and returns an array that contains the results.
+	 * @param callbackfn - A function that accepts the child, its index, and this field.
+	 */
+	mapBoxed<U>(callbackfn: (value: TypedNodeUnion<TTypes>, index: number, array: this) => U): U[];
+
+	readonly length: number;
+
+	// TODO: more and/or better editing APIs. As is, this can't express moves.
+	replaceRange(
+		index: number,
+		count: number,
+		content: Iterable<FlexibleNodeContent<TTypes>>,
+	): void;
+
+	[Symbol.iterator](): Iterator<TypedNodeUnion<TTypes>>;
+
+	// TODO: JS object traversal docs
+	readonly asArray: readonly UnboxNodeUnion<TTypes>[];
+}
+
+/**
+ * Field this stores exactly one child.
+ *
+ * @remarks
+ * Unboxes its content, so in schema aware APIs which do unboxing, the RequiredField itself will be skipped over and its content will be returned directly.
+ * @privateRemarks
+ * Finish renaming from ValueField to RequiredField
```

Comment:
Is this TODO still relevant?

Reply:
Yes. The file kind, schema builder, and probably lots of other things will need renames.

Reply:
Nit
```suggestion
 * TODO: Finish renaming from ValueField to RequiredField
```